drivers/sensor/isl29125/isl29125.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2020 Intive
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+
+#include "isl29125.h"
+
+LOG_MODULE_REGISTER(ISL29125, CONFIG_SENSOR_LOG_LEVEL);
+
+static u8_t isl29125_read8(struct device *i2c_dev, u8_t reg)
+{
+	u8_t buf;
+
+	if (i2c_reg_read_byte(i2c_dev, ISL29125_I2C_ADDR, reg, &buf) < 0) {
+		LOG_ERR("Error reading register %d at Addr:0x%x", reg, ISL29125_I2C_ADDR);
+	}
+	return buf;
+}
+
+static u16_t isl29125_read16(struct device *i2c_dev, u8_t reg)
+{
+	u8_t buf[2];
+
+	if (i2c_write_read(i2c_dev, ISL29125_I2C_ADDR, &reg, 1, &buf, 2) < 0) {
+		LOG_ERR("Error reading register %d at Addr:0x%x", reg, ISL29125_I2C_ADDR);
+	}
+	return sys_le16_to_cpu(buf[0] | (buf[1] << 8));
+}
+
+static void isl29125_write8(struct device *i2c_dev, u8_t reg, u8_t data)
+{
+	if (i2c_reg_write_byte(i2c_dev, ISL29125_I2C_ADDR, reg, data) != 0) {
+		LOG_ERR("Error writting register %d at Addr:0x%x", reg, ISL29125_I2C_ADDR);
+	}
+}
+
+static void isl29125_write16(struct device *i2c_dev, u8_t reg, u16_t data)
+{
+	u8_t buf[3];
+
+	data = sys_cpu_to_le16(data);
+	buf[0] = reg;
+	buf[1] = data & 0xFF;
+	buf[2] = data >> 8;
+	if (i2c_write(i2c_dev, buf, 3, ISL29125_I2C_ADDR)) {
+		LOG_ERR("Error writting register %d at Addr:0x%x", reg, ISL29125_I2C_ADDR);
+	}
+}
+
+static u16_t isl29125_readRed(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_RED_L);
+}
+
+static u16_t isl29125_readGreen(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_GREEN_L);
+}
+
+static u16_t isl29125_readBlue(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_BLUE_L);
+}
+
+static u8_t isl29125_readStatus(struct device *i2c_dev)
+{
+	return isl29125_read8(i2c_dev, ISL29125_STATUS);
+}
+
+static int isl29125_reset(struct device *i2c_dev)
+{
+	u8_t data = 0;
+
+  /* reset to defaults */
```

Comment:
Please replace the two spaces with a tab

Reply:
already done in commit 2e05ee11813